p2p/transport/webrtc/udpmux/muxed_connection.go

```diff
@@ -0,0 +1,164 @@
+package udpmux
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"time"
+
+	pool "github.com/libp2p/go-buffer-pool"
+)
+
+var _ net.PacketConn = &muxedConnection{}
+
+const maxPacketsInQueue int = 128
+
+// muxedConnection provides a net.PacketConn abstraction
+// over packetQueue and adds the ability to store addresses
+// from which this connection (indexed by ufrag) received
+// data.
+type muxedConnection struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	pq     *packetQueue
+	// list of remote addresses associated with this connection.
+	// this is useful as a mapping from [address] -> ufrag
+	addresses []string
```

Comment:
Why is this needed? I thought connections are indexed by their remote address, so a connection should only have a single address, no?